src/structs/neighbor_summary.rs

```diff
@@ -31,18 +34,18 @@ impl NeighborSummary {
         self.num_non_zeros = v;
     }
 
-    pub fn get_horizontal(&self, index: usize) -> i16 {
-        return self.edge_pixels[index + 8];
+    pub fn get_vertical(&self) -> &[i16; 8] {
+        return &self.edge_pixels_v;
     }
 
-    pub fn get_vertical(&self, index: usize) -> i16 {
-        return self.edge_pixels[index];
+    pub fn get_horizontal(&self) -> &[i16; 8] {
```

Comment:
often times making these return by value instead of by reference makes it easier for the optimizer to optimize. Not sure if that holds true here.